util/merkle-tree/src/proof.rs

```diff
@@ -0,0 +1,252 @@
+use crate::hash::HashKernels;
+use crate::tree::Tree;
+use std::collections::VecDeque;
+
+/// Merkle Proof can provide a proof for existence of one or more items.
+/// Only sibling of the nodes along the path that form leaves to root,
+/// excluding the nodes already in the path, should be included in the proof.
+/// For example, if we want to show that [T0, T5] is in the list of 6 items,
+/// only nodes [T4, T1, B3] should be included in the proof.
+///
+/// `tree nodes`: [B0, B1, B2, B3, B4, T0, T1, T2, T3, T4, T5]
+/// `leaves`: [(0, T0), (5, T5)]
+/// `lemmas`: [T4, T1, B3]
+/// `leaves_count`: 6
+pub struct Proof<H>
+where
+    H: HashKernels,
+{
+    /// a partial leaves collection keeps the items sorted based on index
+    pub leaves: Vec<(usize, H::Item)>,
```

Comment:
leaves should not be included

Reply:
will keep it after discussion, unless we find a better naming